### PR TITLE
Fix attested_tls sample build dependencies

### PR DESCRIPTION
**Pull request by jhand2**
Fixes build failure in #2367 by properly expressing build dependencies. More details in the issue.

### CODE REVIEW

**Comment by jhand2** [comment]
bors try

**Comment by jhand2** [comment]
Try out sleep in combo with dependency fix

bors try

**Comment by jhand2** [comment]
bors try-

File: samples/attested_tls/server/enc/CMakeLists.txt [hunk 1]
```diff
@@ -12,6 +12,7 @@ add_custom_command(OUTPUT tls_server_enc.signed tls_server_enc_mrenclave.h
   COMMAND openenclave::oesign sign -e $<TARGET_FILE:tls_server_enc> -c ${CMAKE_SOURCE_DIR}/server/enc/enc.conf -k ${CMAKE_SOURCE_DIR}/server/enc/private.pem
   COMMAND openenclave::oesign dump -e tls_server_enc.signed  > temp.dmp
   COMMAND bash ${CMAKE_SOURCE_DIR}/scripts/gen_mrenclave_header.sh ${CMAKE_SOURCE_DIR}/common/tls_server_enc_mrenclave.h temp.dmp
+  COMMAND ${CMAKE_COMMAND} -E sleep 1
```
Comment (radhikaj) [inline samples/attested_tls/server/enc/CMakeLists.txt]: Does sleeping here ensure that the bash process from the previous line completes?

Comment (jhand2) [reply samples/attested_tls/server/enc/CMakeLists.txt]: yes

**Comment by jhand2** [comment]
@BRMcLaren I ran 6 sandbox builds and they all passed. I feel comfortable marking this as the correct fix.

**Comment by BRMcLaren** [comment]
@jhand2 the fast response and fix is much appreciated!

File: samples/attested_tls/client/enc/CMakeLists.txt [hunk 2]
```diff
@@ -18,7 +18,8 @@ add_executable(tls_client_enc
 	       cert_verifier.cpp
 	       identity_verifier.cpp
 	       ../../common/utility.cpp
-	       ${CMAKE_CURRENT_BINARY_DIR}/tls_client_t.c)
+         ${CMAKE_CURRENT_BINARY_DIR}/tls_client_t.c)
```
Comment (BRMcLaren) [inline samples/attested_tls/client/enc/CMakeLists.txt]: nit: could we realign the dependencies and add a line between them for readability reasons, please.

File: samples/attested_tls/non_enc_client/CMakeLists.txt [hunk 3]
```diff
@@ -7,6 +7,7 @@ add_executable(tls_non_enc_client
 	       client.cpp
 	       verify_callback.cpp
 	       verify_signer_openssl.cpp)
+add_dependencies(tls_non_enc_client tls_server_sign_enc)
```
Comment (BRMcLaren) [inline samples/attested_tls/non_enc_client/CMakeLists.txt]: nit: could we realign the dependencies and add a line between them for readability reasons, please.

**Comment by jhand2** [comment]
@radhikaj Do you have anything else to add before I merge? Thanks!

**Comment by andschwa** [comment]
@jhand2 Can you please put the details in the commit messages instead of just the GitHub issue? That way the rationale can be rediscovered with just the Git repository, and doesn’t depend on GitHub.

**Comment by jhand2** [comment]
> Can you please put the details in the commit messages instead of just the GitHub issue? That way the rationale can be rediscovered with just the Git repository, and doesn’t depend on GitHub.

@andschwa done. Thanks!

**Comment by jhand2** [comment]
@BRMcLaren I will be away from my computer for the rest of the evening. If this is blocking others feel free to merge once @andschwa / @radhikaj  approve.

Thanks!
Jordan

**Pull request merged**